plugins/source/gitlab/client/client.go

```diff
@@ -0,0 +1,80 @@
+package client
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/cloudquery/cloudquery/plugins/source/gitlab/client/services"
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/cloudquery/plugin-sdk/specs"
+	"github.com/rs/zerolog"
+	"github.com/xanzy/go-gitlab"
+)
+
+type Client struct {
+	// This is a client that you need to create and initialize in Configure
+	// It will be passed for each resource fetcher.
+	logger  zerolog.Logger
+	Gitlab  Services
+	spec    specs.Source
+	BaseURL string
+}
+type Services struct {
+	Users    services.UsersClient
+	Groups   services.GroupsClient
+	Projects services.ProjectsClient
+	Settings services.SettingsClient
+	Releases services.ReleasesClient
+}
+
+func (c *Client) Logger() *zerolog.Logger {
+	return &c.logger
+}
+
+func (c *Client) ID() string {
+	return c.spec.Name
+}
+
+func Configure(ctx context.Context, logger zerolog.Logger, s specs.Source) (schema.ClientMeta, error) {
+	gitlabSpec := &Spec{}
+	if err := s.UnmarshalSpec(gitlabSpec); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal gitlab spec: %w", err)
+	}
+
+	gitlabToken := gitlabSpec.Token
+	if gitlabToken == "" {
+		gitlabToken, _ = os.LookupEnv("GITLAB_API_TOKEN")
```

Comment:
Let's have only one way to configure it via the spec given we already have environment and file expansion so it is consistent across all plugins.

Reply:
this way you can also have something like `spec.Validate()` and have the validation in one place